### ISSUE TITLE
Add support for learning rate schedulers

### ISSUE THREAD

**Issue by sebffischer** [mlr-org/mlr3torch#300]

**Comment by cxzhang4** [comment]
Ideas for implementation:

A callback, for example `t_clbk("lr_schedule")`.

It would take a function `schedule`: `epoch` or `batch`, possibly `lr` -> `lr`. And also a string that is either `epoch` or `batch`. Then our implementation would call the function at the appropriate time (e.g. `on_batch_begin`).

[Keras has a callback for this](https://keras.io/api/callbacks/learning_rate_scheduler/)

[D2L has a short section](https://d2l.ai/chapter_optimization/lr-scheduler.html)

[Smith paper on cyclical learning rates](https://arxiv.org/pdf/1506.01186)

Possibly we would need to store the previous `lr`.

We could also offer our own implementations of scheduling functions that are accessible as strings.
